runtime/runtime-params-estimator/emu-cost/README.md

```diff
@@ -36,10 +36,50 @@ Start container and build estimator with:
 
 Now start the estimator under QEMU with the counter plugin enabled (note, that Rust compiler produces SSE4, so specify recent CPU):
 
-    docker> ./emu-cost/counter_plugin/qemu-x86_64 -cpu Westmere-v1 -plugin file=./emu-cost/counter_plugin/libcounter.so ../../target/release/runtime-params-estimator --home /tmp/data --accounts-num 20000 --iters 1 --warmup-iters 1
+    docker> ./emu-cost/counter_plugin/qemu-x86_64 -cpu Westmere-v1 -plugin file=./emu-cost/counter_plugin/libcounter.so \
+         ../../target/release/runtime-params-estimator --home /tmp/data --accounts-num 20000 --iters 1 --warmup-iters 1
 
 Note that it may take some time, as we execute instrumented code under the binary translator.
 
+
+## IO cost calibration
+
+We got to calibrate IO operations cost to instruction counts. Technically instructions and IO are absolutely orthogonal,
+however, as we measure our gas in instructions, we have to compute abstract scaling coefficients binding
+number of bytes read/written in IO to instructions executed.
+We do that by computing following operation:
+
+    ./emu-cost/counter_plugin/qemu-x86_64  -d plugin -cpu Westmere-v1 -plugin file=./emu-cost/counter_plugin/libcounter.so \
+        ../../target/release/genesis-populate --home /tmp/data --additional-accounts-num <NUM_ACCOUNTS>
+
+and checking how much data to be read/written depending on number of create accounts.
+Then we could figure out:
+   * 1 account creation cost in instructions
+   * 1 account creation cost in bytes read and written
+For example, experiments performed in mid Oct 2020 shown the following numbers:
+10M accounts:
+    * 6_817_684_914_212 instructions executed
+    * 168_398_590_013 bytes read
+    * 48_486_537_178 bytes written
+
+Thus 1 account approximately costs:
+    * 681_768 instructions executed
+    * 16840 bytes read
+    * 4849 bytes written
+
+Let's presume that execution, read and write each takes following shares in account cost creation.
+   * Execution: *3/6*
+   * Read: *2/6*
+   * Write: *1/6*
+
+Then we could conclude that:
+   * 1 byte read costs 681768 * 2 / 3 / 16840 = 27 instructions
```

Comment:
Why is this not `681768 * 2 / 6 / 16840`?

Same with the example on the next line.

Reply:
I think it's 2/3 because total cost = execution cost (3/6) + io cost (2/6+1/6), so bytes_read*read_cost_per_inst+bytes_write*write_cost_per_inst should equal to execution cost. Therefore here is 681786 * ((2/6)/(3/6)) / 16840.

Reply:
Yes.